src/validation/user.js

```diff
@@ -0,0 +1,30 @@
+import Joi from 'joi';
+
+export const userRegistrationSchema = Joi.object({
+  username: Joi.string().alphanum().min(2).max(30).required(),
+  email: Joi.string().email({
+    minDomainSegments: 2,
+    tlds: { allow: ['com', 'br', 'net'] } }).required(),
+  password: Joi.string().pattern(new RegExp('^[a-zA-Z0-9]{3,30}$')).required(),
+});
+
+export const loginSchema = Joi.object({
+  user_login: [
+    Joi.string().alphanum().min(2).max(30).required(),
+    Joi.string().email({
+      minDomainSegments: 2,
+      tlds: { allow: ['com', 'br', 'net'] } }).required(),
+  ],
+  password: Joi.string().pattern(new RegExp('^[a-zA-Z0-9]{3,30}$')).required(),
+});
+
+export const changePasswordSchema = Joi.object({
+  id: Joi.number(),
```

Comment:
O id nao está como obrigatório, mas se nao for passado, pelo q eu entendi vai dar pau

Minha opinião
- se o usuário está trocando a propria senha, nao deveria ser necessario ele mandar id no body, e o backend deveria pegar esse id do token
- se um usuário (admin) está trocando a senha de outro usuário, aí passaria o id, mas sera q nao é melhor nao precisar do current_password?